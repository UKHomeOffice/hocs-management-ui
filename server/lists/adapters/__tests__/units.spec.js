const unitsAdapter = require('../units');

const mockLogger = {
    debug: () => { },
    info: () => { },
    warn: () => { },
    error: () => { }
};

describe('Units Adapter', () => {

    it('should transform unit data', async () => {
        const mockData = [
            { displayName: 'displayName1', shortCode: '1', type: 'type1' },
            { displayName: 'displayName2', shortCode: '2', type: 'type2' },
            { displayName: 'displayName2', shortCode: '3', type: 'type3' }
        ];

        const results = await unitsAdapter(mockData, { user: { type: 'type1' }, logger: mockLogger });

        expect(results).toBeDefined();
        expect(results).toMatchSnapshot();
    });
});
