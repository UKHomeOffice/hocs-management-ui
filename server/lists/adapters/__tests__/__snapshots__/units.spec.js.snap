// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Units Adapter should transform unit data 1`] = `
Array [
  Object {
    "displayName": "displayName1",
    "shortCode": "1",
    "type": "type1",
  },
  Object {
    "displayName": "displayName2",
    "shortCode": "2",
    "type": "type2",
  },
  Object {
    "displayName": "displayName2",
    "shortCode": "3",
    "type": "type3",
  },
]
`;
